### PR TITLE
Update UI after link:cancelled event

### PR DESCRIPTION
**Pull request by chloepouprom**
@jugarrit @lemieux @mspensieri @dannytranlx

### CODE REVIEW

File: src/js/components/channels/twilio-channel-content.jsx [hunk 1]
```diff
@@ -26,8 +26,7 @@ export class TwilioChannelContentComponent extends Component {
 
     handleInputChange = (telNumber) => {
         updateTwilioAttributes({
-            appUserNumber: telNumber,
-            hasError: false
```
Comment (chloepouprom) [inline src/js/components/channels/twilio-channel-content.jsx]: I reset the `hasError` flag in `onNumberValid` instead. 

When updating the UI following the `link:cancelled` event, `handleInputChange` would get triggered and would reset the `hasError` flag, thus never rendering the message. 😐

File: src/js/services/integrations-service.js [hunk 2]
```diff
@@ -133,3 +133,16 @@ export function pingTwilioChannel(userId) {
...
         });
 }
+
+export function cancelTwilioLink() {
+    const {user: {pendingClients}, integrations: {twilio: {appUserNumber}}, ui: {text: {smsLinkCancelled}}} = store.getState();
+
+    store.dispatch(updateUser({
+        pendingClients: pendingClients.filter((pendingClient) => pendingClient.platform !== 'twilio')
+    }));
+    updateTwilioAttributes({
```
Comment (lemieux) [inline src/js/services/integrations-service.js]: I'd put a space between those

File: src/js/utils/faye.js [hunk 3]
```diff
@@ -124,6 +125,8 @@ export function handleUserSubscription({appUser, event}) {
                 }, ANIMATION_TIMINGS.PAGE_TRANSITION);
             }, ANIMATION_TIMINGS.PAGE_TRANSITION);
         }
+    } else if (event.type === 'link:cancelled') {
+        return cancelTwilioLink();
```
Comment (lemieux) [inline src/js/utils/faye.js]: I would check the platform before saying it's twilio being cancelled. Sure we only support that for now, but we don't know what the future holds for us and on which version of the widget people will be stuck when we'll roll out new integrations.

**Comment by lemieux** [comment]
:+1:

**Comment by jugarrit** [comment]
👍

**Pull request merged**